XCoreV.core_desc

```diff
@@ -426,59 +426,171 @@ InstructionSet XCoreVMac extends RISCVBase {
             behavior: {
+                X[rd] = (signed)X[rs1] < 0 ? -1 * X[rs1] : X[rs1];  // unsigned
             }
         }
-        CV_CLIPU {
-            encoding: 7'b0111001 :: Iuimm5[4:0] :: rs1[4:0] :: 0b011 :: rd[4:0] :: 0b0101011; // TODO
-            assembly:"{name(rd)}, {name(rs1)}, {name(rs2)}";  // TODO
+        CV_ADDN {
+            encoding: 2'b00 :: Is3[4:0] :: rs2[4:0] :: rs1[4:0] :: 3'b010 :: rd[4:0] :: 7'b1011011;
+            assembly:"{cv.addn}, {name(rd)}, {name(rs1)}, {name(rs2)}, {Is3}";
             behavior: {
+              X[rd] = ((signed)(X[rs1] + X[rs2])) >> Is3;
             }
         }
-        CV_ADDN {
-            encoding: 2'b00 :: Is3[4:0] :: rs2[4:0] :: rs1[4:0] :: 3'b010 :: rd[4:0] :: 7'b1011011;
-            assembly:"{name(rd)}, {name(rs1)}, {name(rs2)}, {Is3}";
+        CV_ADDNR {
+            encoding: 7'b1000000 :: rs2[4:0] :: rs1[4:0] :: 3'b011 :: rd[4:0] :: 7'b0101011;
+            assembly:"{cv.addnr}, {name(rd)}, {name(rs1)}, {name(rs2)}";
+            behavior: {
+               X[rd] = ((signed)(X[rd] + X[rs1])) >> (unsigned)X[rs2][4:0];
+            }
+        }
+        CV_ADDRN {
+            encoding: 2'b10 :: Is3[4:0] :: rs2[4:0] :: rs1[4:0] :: 3'b010 :: rd[4:0] :: 7'b1011011;
+            assembly:"{cv.addrn}, {name(rd)}, {name(rs1)}, {name(rs2)}, {Is3}";
             behavior: {
-              // unsigned<32> result = ((unsigned)X[rs1][15:0] + (unsigned)X[rs2][15:0]) >> Is3;
-              // if(rd != 0) X[rd] = result[31:0];
+               X[rd] = ((signed)(X[rs1] + X[rs2] + 2^(Is3 - 1))) >> Is3;
             }
         }
         CV_ADDUN {
             encoding: 2'b01 :: Is3[4:0] :: rs2[4:0] :: rs1[4:0] :: 3'b010 :: rd[4:0] :: 7'b1011011;
-            assembly:"{name(rd)}, {name(rs1)}, {name(rs2)}, {Is3}";
+            assembly:"{cv.addun}, {name(rd)}, {name(rs1)}, {name(rs2)}, {Is3}";
+            behavior: {
+               X[rd] = (unsigned)(X[rs1] + X[rs2]) >> Is3;
+            }
+        }
+        CV_ADDUNR {
+            encoding: 7'b1000001 :: rs2[4:0] :: rs1[4:0] :: 3'b011 :: rd[4:0] :: 7'b0101011;
+            assembly:"{cv.addunr}, {name(rd)}, {name(rs1)}, {name(rs2)}";
+            behavior: {
+               X[rd] = (unsigned)(X[rd] + X[rs1]) >> X[rs2][4:0];
+            }
+        }
+        CV_ADDURN {
+            encoding: 2'b11 :: Is3[4:0] :: rs2[4:0] :: rs1[4:0] :: 3'b010 :: rd[4:0] :: 7'b1011011;
+            assembly:"{cv.addurn}, {name(rd)}, {name(rs1)}, {name(rs2)}, {Is3}";
+            behavior: {
+               X[rd] = (unsigned)(X[rs1] + X[rs2] + 2^(Is3 - 1)) >> Is3;
+            }
+        }
+        CV_CLIP {
+            encoding: 7'b0111000 :: Is2[4:0] :: rs1[4:0] :: 3'b011 :: rd[4:0] :: 7'b0101011;
+            assembly:"{cv.clip}, {name(rd)}, {name(rs1)}, {Is2}";
+            behavior: {
+                X[rd] = X[rs1]<= (-2)^(Is2 - 1) ? (-2)^(Is2 - 1) : (rs1 >= (2)^(Is2 - 1) ? (2)^(Is2 - 1) : X[rs1]);
```

Comment:
@khouly18 could you please add more parentheses here (also for the other CLIP* instructions)